### PR TITLE
Make tetramm scale averaging rate with frame time

### PR DESCRIPTION
**Pull request by callumforrester**
Fixes https://github.com/DiamondLightSource/i22-bluesky/issues/60

### Instructions to reviewer on how to test:
1. Confirm tests pass

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

**Comment by stan-dot** [comment]
I have some issues re-running the tests, might be about the dependencies

**Comment by stan-dot** [comment]
had to update ophyd-async.core

**Review by stan-dot — APPROVED** [review]
nit: I might do `exposure_seconds` or `exposure_time` instead of just `exposure`

**Comment by callumforrester** [comment]
I sort-of agree, but `exposure` is ophyd-async's term and I'd rather consistency in this case

File: tests/devices/unit_tests/test_tetramm.py [hunk 1]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
 async def test_max_frame_rate_is_calculated_correctly(
     tetramm_controller: TetrammController,
 ):
-    tetramm_controller.minimum_frame_time = 2.0
+    status = await tetramm_controller.arm(1, DetectorTrigger.edge_trigger, 2.0)
+    await status
 
-    assert tetramm_controller.minimum_frame_time == 0.1
+    assert tetramm_controller.minimum_expsoure == 0.1
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:61]:
```suggestion
    assert tetramm_controller.minimum_exposure == 0.1
```

File: tests/devices/unit_tests/test_tetramm.py [hunk 2]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
     assert tetramm_controller.max_frame_rate == 10.0
 
     # Ensure that the minimum frame time is correctly calculated given a maximum
     # frame rate.
-    # max_frame_rate**-1 = minimum_frame_times
+    # max_frame_rate**-1 = minimum_expsoures
     tetramm_controller.max_frame_rate = 20.0
-    assert tetramm_controller.minimum_frame_time == pytest.approx(1 / 20)
+    assert tetramm_controller.minimum_expsoure == pytest.approx(1 / 20)
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:68]:
```suggestion
    assert tetramm_controller.minimum_exposure == pytest.approx(1 / 20)
```

File: tests/devices/unit_tests/test_tetramm.py [hunk 3]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
     # Using coprimes to ensure the solution has a unique relation to the values.
     tetramm_controller.base_sample_rate = 100_000
     tetramm_controller.readings_per_frame = 999
     tetramm_controller.maximum_readings_per_frame = 1_001
     tetramm_controller.minimum_values_per_reading = 17
 
-    # min_frame_time (s/f) = max_readings_per_frame * values_per_reading / sample_rate (v/s)
-    minimum_frame_time = (
+    # min_expsoure (s/f) = max_readings_per_frame * values_per_reading / sample_rate (v/s)
+    minimum_expsoure = (
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:81]:
```suggestion
    minimum_exposure = (
```

File: tests/devices/unit_tests/test_tetramm.py [hunk 4]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
         tetramm_controller.readings_per_frame
         * tetramm_controller.minimum_values_per_reading
         / float(tetramm_controller.base_sample_rate)
     )
 
-    assert tetramm_controller.minimum_frame_time == pytest.approx(minimum_frame_time)
+    assert tetramm_controller.minimum_expsoure == pytest.approx(minimum_expsoure)
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:87]:
```suggestion
    assert tetramm_controller.minimum_exposure == pytest.approx(minimum_exposure)
```

File: tests/devices/unit_tests/test_tetramm.py [hunk 5]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
+    status = await tetramm_controller.arm(1, DetectorTrigger.edge_trigger, 0.1)
+    await status
     assert tetramm_controller.readings_per_frame == int(readings_per_time * 0.1)
 
 
 VALID_TEST_EXPOSURE_TIME = 1 / 19
 
 
-async def test_set_frame_time_updates_values_per_reading(
+async def test_set_expsoure_updates_values_per_reading(
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:120]:
```suggestion
async def test_set_exposure_updates_values_per_reading(
```

File: tests/devices/unit_tests/test_tetramm.py [hunk 6]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
     tetramm_controller: TetrammController,
     tetramm_driver: TetrammDriver,
 ):
-    await tetramm_controller.set_frame_time(VALID_TEST_EXPOSURE_TIME)
+    await tetramm_controller.set_expsoure(VALID_TEST_EXPOSURE_TIME)
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:124]:
```suggestion
    await tetramm_controller.set_exposure(VALID_TEST_EXPOSURE_TIME)
```

File: tests/devices/unit_tests/test_tetramm.py [hunk 7]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
     values_per_reading = await tetramm_driver.values_per_reading.get_value()
     assert values_per_reading == 5
 
 
-async def test_set_invalid_frame_time_for_number_of_values_per_reading(
+async def test_set_invalid_expsoure_for_number_of_values_per_reading(
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:129]:
```suggestion
async def test_set_invalid_exposure_for_number_of_values_per_reading(
```

File: tests/devices/unit_tests/test_tetramm.py [hunk 8]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
     tetramm_controller: TetrammController,
 ):
     """
-    frame_time >= readings_per_frame * values_per_reading / sample_rate
+    expsoure >= readings_per_frame * values_per_reading / sample_rate
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:133]:
```suggestion
    exposure >= readings_per_frame * values_per_reading / sample_rate
```

File: tests/devices/unit_tests/test_tetramm.py [hunk 9]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
     With the default values:
     base_sample_rate = 100_000
     minimum_values_per_reading = 5
     readings_per_frame = 1_000
-    frame_time >= 1_000 * 5 / 100_000 = 1/20
+    expsoure >= 1_000 * 5 / 100_000 = 1/20
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:138]:
```suggestion
    exposure >= 1_000 * 5 / 100_000 = 1/20
```

**Review by DiamondJoseph — CHANGES_REQUESTED** [review]
Bunch of nits, but hopefully should be easy to apply with plugin

Logic looks correct?

**Comment by DiamondJoseph** [comment]
I agree that exposure is standard, when spelled correctly ;)

Call out the units in the docstring if required.

File: tests/devices/unit_tests/test_tetramm.py [hunk 10]
```diff
@@ -55,91 +55,125 @@ async def tetramm(static_directory_provider: DirectoryProvider) -> TetrammDetect
...
     """
-    frame_time >= readings_per_frame * values_per_reading / sample_rate
+    exposure >= readings_per_frame * values_per_reading / sample_rate
     With the default values:
     base_sample_rate = 100_000
     minimum_values_per_reading = 5
     readings_per_frame = 1_000
-    frame_time >= 1_000 * 5 / 100_000 = 1/20
+    exposure >= 1_000 * 5 / 100_000 = 1/20
     """
```
Comment (DiamondJoseph) [inline tests/devices/unit_tests/test_tetramm.py:139]: nit: this docstring isn't relevant anymore to the numbers being used

**Review by DiamondJoseph — APPROVED** [review]

**Pull request merged**